cmd/kube-apiserver/app/server.go

```diff
@@ -627,8 +627,14 @@ func defaultOptions(s *options.ServerRunOptions) error {
 	if err := s.SecureServing.MaybeDefaultWithSelfSignedCerts(s.GenericServerRunOptions.AdvertiseAddress.String(), []string{"kubernetes.default.svc", "kubernetes.default", "kubernetes"}, []net.IP{apiServerServiceIP}); err != nil {
 		return fmt.Errorf("error creating self-signed certificates: %v", err)
 	}
-	if err := s.CloudProvider.DefaultExternalHost(s.GenericServerRunOptions); err != nil {
-		return fmt.Errorf("error setting the external host value: %v", err)
+	if len(s.GenericServerRunOptions.ExternalHost) != 0 {
```

Comment:
this is backwards... should be `if len(s.GenericServerRunOptions.ExternalHost) == 0`

Reply:
Done